### ISSUE TITLE
Score-specific configuration option for meaning of accidentals

### ISSUE THREAD

**Issue by basil** [danielgarthur/neanes#331]
See [this thread](https://analogion.com/forum/index.php?threads/confusion-with-sharps-%CE%94%CE%B9%CE%AD%CF%83%CE%B5%CE%B9%CF%82-and-flats-%CE%A5%CF%86%CE%AD%CF%83%CE%B5%CE%B9%CF%82.11843/). A while back, Papa Ephraim researched this and came to [the following conclusion](https://music.samonastery.org/BrailleNeumes.html):

> Unfortunately, some confusion currently prevails regarding the exact meaning of some sharps and flats. When Chrysanthos devised the "New Method" of Byzantine Music notation in the early 19th century, he decided to use five different symbols for sharps and five for flats. The amount that each of those symbols changes the pitch is proportional to the interval between it and the next note. Thus, a plain sharp (i.e., a sharp without a hook or crossbeams) raises the pitch halfway to the next note in the scale. If that interval is a whole step (12 units—or 200 cents), that sharp will raise the pitch by 6 units (100 cents). But if that interval is only 10 units (167 cents), it will raise the pitch by only 5 units (83 cents).
>
> In order to make Byzantine Music notation more definitive, the Patriarchal Committee of 1881 decided to change the meaning of sharps and flats. They eliminated sharps and flats that have a hook, and they used sharps and flats with 0, 1, 2, 3, or 4 crossbeams. The meaning they assigned to these symbols was as follows: A plain sharp (i.e., without any crossbeams) adds 2 units (33 cents) to the pitch of a note, and each additional crossbeam added to that sharp adds another 2 units to the pitch. Thus, a sharp with one crossbeam adds 4 units (67 cents), a sharp with two crossbeams adds 6 units (100 cents), a sharp with three crossbeams adds 8 units (133 cents), and a sharp with four crossbeams adds 10 units (167 cents).
>
> The problem is that many books printed after 1881 continued using a plain sharp (i.e., without any crossbeams) not with the new meaning assigned to this symbol by the 1881 Committee (which stated that it adds two units or 33 cents to the pitch), but with the meaning given to it by Chrysanthos (who stated that it increases the note's pitch halfway to the next higher note).

I am preparing a collection of 60+ Axion Estins, covering a stylistically broad spectrum of time periods and compositional styles, so I have had to contend with this issue. Most classical scores were published before 1881 and use the older convention, where most sharps/flats don't have a crossbeam and typically mean a change of 5-6 moria. Even many books printed after 1881 continue to follow this older convention for classical pieces. Most of the time when you see the newer convention with the crossbeams, it is either (a) in a modern composition or (b) in classical-style pieces from members of the Karas school.

Someone like me who is striving for editorial consistency has three options:

- Follow the newer convention with many crossbeams for all pieces, both classical-style and modern-style. This effectively means updating the classical-style pieces to use the crossbeams, as members of the Karas school often do. I'm not a fan of this choice, both because I think it unnecessarily clutters the classical scores and because some of the interval choices of the Karas school are controversial.
- Follow the older convention without many crossbeams for all pieces, both classical-style and modern-style. This effectively means stripping the modern-style pieces of many crossbeams. I'm not a fan of this either, because it unnecessarily lowers the granularity of the notation for modern-style pieces. These scores are typically fairly cluttered in the first place, so adding a few crossbeams doesn't hurt.
- Compromise by following a different convention based on the type of piece and its intended performance style.

In my opinion the software should support both conventions. Those who want to continue publishing the classical-style pieces in their original format shouldn't be forced to update the accidentals to get quality playback. Those who want to update the conventions of the classical-style pieces should be free to do so, but this controversial practice shouldn't be forced onto everyone. Those who want to publish modern-style pieces with lots of notational clutter and fine-grained accidentals should also be free to do so.

In short, I think the software should have a configuration preference. That preference should not only impact playback, but also it should "stick" to the score when the score is saved, so that it is possible to load two scores, each with different conventions, and play them back correctly without having to change the playback settings each time.

**Comment by basil** [comment]
I can likely implement the playback portion of this after #322 is merged, but I would need some guidance as to how we want to expose this option in the frontend and/or on-disk file format.
